### PR TITLE
Do not allow WCSes to be reverse-indexed

### PR DESCRIPTION
**Pull request by keflavich**
Addresses https://github.com/astropy/astropy/issues/4529

It would be nice to have access to this sort of functionality, but it requires having access to NAXIS to compute correctly (see https://github.com/astropy/astropy/issues/4669).

### CODE REVIEW

**Comment by pllim** [comment]
Tests failed.

File: astropy/wcs/wcs.py [hunk 1]
```diff
@@ -2919,6 +2919,9 @@ def slice(self, view, numpy_order=True):
 
         wcs_new = self.deepcopy()
         for i, iview in enumerate(view):
+            if iview.step < 0:
```
Comment (astrofrog) [inline astropy/wcs/wcs.py]: @keflavich - I think you probably want something like ``if iview.step is not None and iview.step < 0``

**Comment by astrofrog** [comment]
@keflavich - can you add a changelog entry in 1.2? (in bug fixes I think)

File: astropy/utils/console.py [hunk 2]
```diff
@@ -165,6 +165,12 @@ def terminal_size(file=None):
         if width > 10:
             width -= 1
         return (lines, width)
+    except KeyboardInterrupt:
+        # anonymous exception catching will catch - and therefore prevent -
+        # keyboard interrupts, which means users can't manually break out of
+        # loops.  Catching and raising KeyboardInterrupts allows all other
+        # exceptions to pass into the next clause below
+        raise
```
Comment (astrofrog) [inline astropy/utils/console.py]: @keflavich - I think you added this by mistake - can you remove and rebase to clean the history?

**Pull request merged**